tools/dotnet-scaffold-aspire/Commands/StorageCommand.cs

```diff
@@ -1,51 +1,232 @@
+            UsingTopLevelsStatements = await ProjectModifierHelper.IsUsingTopLevelStatements(codeService)
+        };
 
-        [CommandOption("--prerelease")]
-        public required bool Prerelease { get; set; }
+        //edit CodeModifierConfig to add the web project name from _autoGeneratedProjectNames.
+        _autoGeneratedProjectNames.TryGetValue(commandSettings.Project, out var autoGenProjectName);
+        config = EditConfigForAppHost(config, options, autoGenProjectName, commandSettings.Type);
+
+        var projectModifier = new ProjectModifier(
+            _environmentService,
+            hostAppSettings,
+            codeService,
+            _logger,
+            options,
+            config);
+        return await projectModifier.RunAsync();
+    }
+
+    internal async Task GetAutoGeneratedProjectNamesAsync(CodeService codeService, string projectPath)
+    {
+        var allDocuments = await codeService.GetAllDocumentsAsync();
+        var allSyntaxRoots = await Task.WhenAll(allDocuments.Select(doc => doc.GetSyntaxRootAsync()));
+
+        // Get all classes with the "Projects" namespace
+        var classesInNamespace = allSyntaxRoots
+            .SelectMany(root => root?.DescendantNodes().OfType<ClassDeclarationSyntax>() ?? Enumerable.Empty<ClassDeclarationSyntax>())
+            .Where(cls => cls.IsInNamespace("Projects"))
+            .ToList();
+
+        foreach (var classSyntax in classesInNamespace)
+        {
+            string? projectPathValue = classSyntax.GetStringPropertyValue("ProjectPath");
+            // Get the full class name including the namespace
+            var className = classSyntax.Identifier.Text;
+            if (!string.IsNullOrEmpty(projectPathValue))
+            {
+                _autoGeneratedProjectNames.Add(projectPathValue, $"Projects.{className}");
+            }
+        }
+    }
+
+    internal async Task<bool> UpdateWebApiAsync(StorageCommandSettings commandSettings)
+    {
+        if (!MSBuildLocator.IsRegistered)
+        {
+            MSBuildLocator.RegisterDefaults();
+        }
+
+        CodeModifierConfig? config = ProjectModifierHelper.GetCodeModifierConfig("storage-webapi.json", System.Reflection.Assembly.GetExecutingAssembly());
+        var workspaceSettings = new WorkspaceSettings
+        {
+            InputPath = commandSettings.Project
+        };
+
+        var webApiSettings = new AppSettings();
+        webApiSettings.AddSettings("workspace", workspaceSettings);
+        var codeService = new CodeService(webApiSettings, _logger);
+
+        CodeChangeOptions options = new()
+        {
+            IsMinimalApp = await ProjectModifierHelper.IsMinimalApp(codeService),
+            UsingTopLevelsStatements = await ProjectModifierHelper.IsUsingTopLevelStatements(codeService)
+        };
+
+        config = EditConfigForApiService(config, options, commandSettings.Type);
+        var projectModifier = new ProjectModifier(
+        _environmentService,
```

Comment:
nit: indentation (this line and following 5)

Reply:
fixed tyty